### PR TITLE
Create new json objects for service models

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/dotnet/tye/issues/97. Fixes https://github.com/dotnet/tye/issues/98.

I'm still using the OM to get the dashboard URL. Other notes inline.

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Tye.Hosting.Model
+{
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs]: I'm not a fan of just appending JSON to the end of these names. Anything better?

File: src/Microsoft.Tye.Hosting/Model/ServiceDescription.cs [hunk 2]
```diff
@@ -25,13 +25,7 @@ public ServiceDescription(string name, RunInfo? runInfo)
...
     {
-        public ConfigurationSource(string name, string value)
-        {
-            Name = name;
-            Value = value;
-        }
-
-        public string Name { get; }
-        public string Value { get; }
+        public string? Name { get; set; }
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/Model/ServiceDescription.cs]: Organizationally, these are objects that are used by the JSON and non-JSON. I left them in place for now.

File: src/Microsoft.Tye.Hosting/Model/JSON/ServiceDescriptionJson.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Tye.Hosting.Model
+{
+    public class ServiceDescriptionJson
+    {
+        public string? Name { get; set; }
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/Model/JSON/ServiceDescriptionJson.cs]: I realize I'm currently not serializing/deserializing RunInfo. I'll take a stab at it now, but it's a bit tricky with all the subclasses.

File: src/Microsoft.Tye.Hosting/Model/Application.cs [hunk 4]
```diff
@@ -30,7 +30,7 @@ public void PopulateEnvironment(Service service, Action<string, string> set, str
                 // Inject normal configuration
                 foreach (var pair in service.Description.Configuration)
                 {
-                    set(pair.Name, pair.Value);
+                    set(pair.Name ?? "", pair.Value ?? "");
```
Comment (rynowak) [inline src/Microsoft.Tye.Hosting/Model/Application.cs]: Is this right? is a empty configuration value valid? or should we skip entries with null?

File: src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs (hunk 1 above)
Comment (rynowak) [reply src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs]: I'd use a separate namespace and possibly prepend V1 to all of the things that are serialization DTOs.

File: src/Microsoft.Tye.Hosting/Model/Application.cs (hunk 4 above)
Comment (jkotalik) [reply src/Microsoft.Tye.Hosting/Model/Application.cs]: Yeah I'll skip it, I don't think these will be null but might as well prevent it.

File: src/Microsoft.Tye.Hosting/Model/PortMapping.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+using System.Collections.Generic;
+
+namespace Microsoft.Tye.Hosting.Model
+{
+    public class PortMapping
+    {
+        public int ExternalPort { get; set; }
+
+        public List<int> InteralPorts { get; set; } = new List<int>();
```
Comment (rynowak) [inline src/Microsoft.Tye.Hosting/Model/PortMapping.cs]: InternalPorts

File: src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.Tye.Hosting/Model/JSON/ReplicaStatusJson.cs]: Alright, I'll fully commit to it.

**Pull request merged**